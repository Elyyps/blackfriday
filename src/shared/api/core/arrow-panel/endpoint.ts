import { generateArrowPanelData } from "./generate-dummy-data";

export const generateTabsDummyData = generateArrowPanelData();
