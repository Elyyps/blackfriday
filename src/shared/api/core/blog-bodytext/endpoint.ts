import { generateDummyBlogBodyText, generateBlackFriday2019Text } from "./generate-dummy-data";

export const generateBlogBodyTextData = generateDummyBlogBodyText;
export const generateBlackFriday2019TextData = generateBlackFriday2019Text;
