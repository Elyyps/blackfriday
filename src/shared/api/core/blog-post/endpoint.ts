import { generateDummyBlogPost, generate2DummyBlogPosts } from "./generate-dummy-data";

export const generateBlogPostData = generateDummyBlogPost;
export const generate2BlogPostsData = generate2DummyBlogPosts;
