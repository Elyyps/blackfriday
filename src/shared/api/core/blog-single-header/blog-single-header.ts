import { Link } from "../link";

export class BlogSingleHeader {
  public button: Link;
  public image: string;
  public links: Link[];
  public title: string;
}
