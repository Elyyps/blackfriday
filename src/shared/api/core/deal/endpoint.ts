import { Deal } from "./deal";
import { generateDemoDealDummyData } from "./generate-dummy-data";
import { FilterItem } from "../filter/filter-item";
import { shuffle } from "@app/util/array";
/* tslint:disable */
export interface IDealResult {
  deals: Deal[];
  totalResults: number;
}

const initialStoresResult = [...generateDemoDealDummyData()];

const getStores = (
  skip: number,
  take: number,
  storeFilters: FilterItem[],
  categoryFilters: FilterItem[],
  brandFilters: FilterItem[],
  sortBy?: string
): IDealResult => {
  const selectedCategoryFilters = getSelectedFilters(categoryFilters);
  const selectedBrandFilters = getSelectedFilters(brandFilters);
  const selectedStoreFilters = getSelectedFilters(storeFilters);

  let result = [...initialStoresResult];

  if (selectedCategoryFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.categories.filter(category => {
        return selectedCategoryFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedBrandFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedBrandFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedStoreFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedStoreFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  const totalResults = result.length;

  result = result.splice(skip, take);

  switch (sortBy) {
    case "Relevantie":
      // Keep default sorting
      break;
    case "Nieuwste":
      result.reverse();
      break;
    case "Populair":
      result = shuffle(result);
      break;
    case "Winkels A - Z":
      result = result.sort((a, b) => {
        if (a.name < b.name) {
          return -1;
        }
        if (b.name < a.name) {
          return 1;
        }

        return 0;
      });
      break;
    case "Winkels Z - A":
      result = result.sort((a, b) => {
        if (a.name > b.name) {
          return -1;
        }
        if (b.name > a.name) {
          return 1;
        }

        return 0;
      });
      break;
    default:
      /* tslint:disable:switch-final-break  */
      // keep default sorting
      break;
  }

  return {
    deals: result,
    totalResults
  };
};

const getSelectedFilters = (filterItems: FilterItem[]): string[] | undefined => {
  const selectedFilterItems = filterItems.filter(item => item.isSelected);

  if (selectedFilterItems && selectedFilterItems.length > 0) {
    return selectedFilterItems.map(item => item.displayName);
  }

  return undefined;
};

const getAmountForCategory = (category: string): number =>
  initialStoresResult.filter(deal => deal.categories.indexOf(category) > -1).length;

const getAmountForBrand = (brand: string): number =>
  initialStoresResult.filter(deal => deal.availableBrands.indexOf(brand) > -1).length;

const getAmountForStore = (store: string): number =>
  initialStoresResult.filter(deal => deal.stores.indexOf(store) > -1).length;

export const DealApi = {
  getStores,
  initialStoresResult,
  getAmountForCategory,
  getAmountForBrand,
  getAmountForStore
};
