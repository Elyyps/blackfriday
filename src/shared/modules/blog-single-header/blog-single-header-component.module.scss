.blog-single-header {
  &-basic {
    color: $global-inverse-color;
    text-align: center;
    position: relative;
    z-index: 1;
    margin-bottom: 40px;
    padding: 72px 0;
    @media (max-width: $breakpoint-small-max) {
      margin-bottom: 24px;
      padding: 40px 0;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      text-align: left;
    }
    &::before,
    &::after {
      position: absolute;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      content: "";
      background: $gray-dark;
      z-index: -1;
    }
    &::after {
      background-position: 50% 0;
      background-size: 2000px auto;
      opacity: 0.5;
      background: url("https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/pattern.svg");
    }

    &__breadcrumbs {
      font-size: 12px;
      font-weight: 600;
      color: #999;
      display: flex;
      justify-content: center;
      margin-bottom: 8px;
      li {
        margin: 0 5px;
        display: inline-flex;
        align-items: center;
        &:not(:last-child) {
          &::after {
            margin-left: 8px;
            display: block;
            content: "\2014";
          }
        }
      }
      a {
        color: #999;
        text-decoration: none;
        &:hover {
          text-decoration: none;
          color: $global-inverse-color;
        }
      }
    }

    &__back {
      color: #999;
      text-decoration: none;
      font-size: 12px;
      font-weight: 600;
      display: flex;
      align-items: center;
      margin-bottom: 10px;
      &:hover {
        text-decoration: none;
        color: $global-inverse-color;
      }
      svg {
        margin-right: 10px;
        path {
          stroke: currentColor;
        }
      }
    }

    h1 {
      font-weight: bold;
      margin: 0;
      color: currentColor;
    }
  }
  position: relative;
  z-index: 1;
  margin-bottom: 40px;
  @media (max-width: $breakpoint-small-max) {
    margin-bottom: 24px;
  }
  &::before,
  &::after {
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 150px;
    content: "";
    background: $gray-dark;
    z-index: -1;
    @media (max-width: $breakpoint-small-max) {
      bottom: 70px;
    }
  }
  &::after {
    background: url("https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/pattern.svg");
    background-position: 50% 0;
    background-size: 2000px auto;
    opacity: 0.5;
  }
  &__image {
    height: 400px;
    border-radius: 2px;
    overflow: hidden;
    position: relative;
    @media (max-width: $breakpoint-small-max) {
      height: 250px;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      height: 160px;
    }
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &-basic {
    margin: 0;
    &::before,
    &::after {
      display: none;
    }
  }
}
