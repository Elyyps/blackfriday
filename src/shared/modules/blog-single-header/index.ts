export * from "./blog-single-header.component";
