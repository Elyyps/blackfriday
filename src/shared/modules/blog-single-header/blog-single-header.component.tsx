import * as React from "react";
import styles from "./blog-single-header-component.module.scss";
import { ImageComponent } from "../../core/image";
import { IconComponent } from "../../core/icon";
import { Link } from "react-router-dom";
import IconBack from "@assets/icons/chevron-left.svg";
import { BlogSingleHeaderModule } from "@app/api/modules/blog-single-header/blog-single-header";

export interface IBlogSingleHeaderComponentProps {
  blogSingleHeader: BlogSingleHeaderModule;
}

const BlogSingleHeaderComponent = (props: IBlogSingleHeaderComponentProps) => (
  <div className={styles["blog-single-header"]}>
    <div className={styles["blog-single-header-basic"]}>
      <div className="uk-container">
        <ul className={` "uk-visible@s" ${styles["blog-single-header-basic__breadcrumbs"]}`}>
          {props.blogSingleHeader.links &&
            props.blogSingleHeader.links.map((link, key) => (
              <li key={key}>
                <Link to={link.url}>{link.title}</Link>
              </li>
            ))}
        </ul>

        {props.blogSingleHeader.button && (
          <Link
            className={`"uk-hidden@s" ${styles["blog-single-header-basic__back"]}`}
            to={props.blogSingleHeader.button.url}
          >
            <IconComponent icon={IconBack} size={"6px"} />
            {props.blogSingleHeader.button.title}
          </Link>
        )}

        {props.blogSingleHeader.title && <h1>{props.blogSingleHeader.title}</h1>}
      </div>
    </div>
    <div className="uk-container">
      <div className={styles["blog-single-header__image"]}>
        <ImageComponent src={props.blogSingleHeader.image} />
      </div>
    </div>
  </div>
);

export { BlogSingleHeaderComponent };
