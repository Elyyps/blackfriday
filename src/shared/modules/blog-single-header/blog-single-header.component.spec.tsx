import React from "react";
import { shallow } from "enzyme";
import { BlogSingleHeaderComponent } from "./blog-single-header.component";
import { generateBlogSingleHeaderData } from "@app/api/modules/blog-single-header/endpoint";
const data = generateBlogSingleHeaderData();
describe("[BlogSingleHeader]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<BlogSingleHeaderComponent blogSingleHeader={data} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
