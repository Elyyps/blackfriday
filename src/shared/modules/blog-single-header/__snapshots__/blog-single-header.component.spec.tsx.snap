// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BlogSingleHeader] should render component without crashing 1`] = `
<div
  className="blog-single-header"
>
  <div
    className="header-content"
  >
    <div
      className="uk-container"
    >
      <HeaderContentComponent
        breadcrumbProps={
          Object {
            "backButton": Object {
              "title": "Terug",
              "url": "/#",
            },
            "links": Array [
              Object {
                "title": "Home",
                "url": "/",
              },
              Object {
                "title": "Alle blogs",
                "url": "#",
              },
            ],
            "variant": "light",
          }
        }
        headerMobilePosition="left"
        title="Vijf voorspellingen van Black Friday Nederland"
      />
    </div>
  </div>
  <div
    className="uk-container"
  >
    <div
      className="image"
    >
      <ImageComponent
        isBlocking={true}
        src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/blog/2.png"
      />
    </div>
  </div>
</div>
`;
